sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -61,7 +61,6 @@ case class CreateArray(children: Seq[Expression]) extends Expression {
     ctx.addMutableState("Object[]", values, s"this.$values = null;")
 
     ev.copy(code = s"""
-      final boolean ${ev.isNull} = false;
```

Comment:
I removed the line because the variable will never be used.
The generated code of the caller of this expression will use literal `false` and Janino might be able to optimize the conditional expressions or if/else statements for null checking.